src/openfermion/utils/_unitary_cc.py

```diff
@@ -0,0 +1,199 @@
+        double_amplitudes_list.append([[i, j, k, l],
+                                      double_amplitudes[i, j, k, l]])
+    return single_amplitudes_list, double_amplitudes_list
+
+
+def uccsd_singlet_paramsize(n_qubits, n_electrons):
+    """Determine number of independent amplitudes for singlet UCCSD
+
+    Args:
+        n_qubits(int): Number of qubits/spin-orbitals in the system
+        n_electrons(int): Number of electrons in the reference state
+
+    Returns:
+        Number of independent parameters for singlet UCCSD with a single
+        reference.
+    """
+    n_occupied = int(numpy.ceil(n_electrons / 2.))
+    n_virtual = n_qubits / 2 - n_occupied
+
+    n_single_amplitudes = n_occupied * n_virtual
+    n_double_amplitudes = n_single_amplitudes ** 2
+    return (n_single_amplitudes + n_double_amplitudes)
+
+
+def uccsd_singlet_operator(packed_amplitudes,
+                           n_qubits,
+                           n_electrons):
+    """Create a singlet UCCSD generator for a system with n_electrons
+
+    This function generates a FermionOperator for a UCCSD generator designed
+        to act on a single reference state consisting of n_qubits spin orbitals
+        and n_electrons electrons, that is a spin singlet operator, meaning it
+        conserves spin.
+
+    Args:
+        packed_amplitudes(ndarray): Compact array storing the unique single
+            and double excitation amplitudes for a singlet UCCSD operator.
+            The ordering lists unique single excitations before double
+            excitations.
+        n_qubits(int): Number of spin-orbitals used to represent the system,
+            which also corresponds to number of qubits in a non-compact map.
+        n_electrons(int): Number of electrons in the physical system.
+
+    Returns:
+        uccsd_generator(FermionOperator): Generator of the UCCSD operator that
+            builds the UCCSD wavefunction.
+    """
+    n_occupied = int(numpy.ceil(n_electrons / 2.))
+    n_virtual = int(n_qubits / 2 - n_occupied)  # Virtual Spatial Orbitals
+    n_t1 = int(n_occupied * n_virtual)
+
+    t1 = packed_amplitudes[:n_t1]
+    t2 = packed_amplitudes[n_t1:]
+
+    def t1_ind(i, j):
+        return i * n_occupied + j
+
+    def t2_ind(i, j, k, l):
+        return (i * n_occupied * n_virtual * n_occupied +
+                j * n_virtual * n_occupied +
+                k * n_occupied +
+                l)
+
+    uccsd_generator = FermionOperator()
+
+    spaces = range(n_virtual), range(n_occupied), range(2)
+
+    for i, j, s in itertools.product(*spaces):
+        uccsd_generator += FermionOperator(
+            (
+                (2 * (i + n_occupied) + s, 1),
+                (2 * j + s, 0),
+            ),
+            coefficient=t1[t1_ind(i, j)])
+
+        uccsd_generator += FermionOperator(
+            (
+                (2 * j + s, 1),
+                (2 * (i + n_occupied) + s, 0),
+            ),
+            coefficient=-t1[t1_ind(i, j)])
```

Comment:
I'm pretty sure that this right here is a pep8 violation. Run the linter?

Reply:
Passes just fine on mine with pep8, I'll add some spaces if that's what you mean?

Reply:
Oops, yeah you're good. Its because that equal sign is setting the value of an optional parameter. That is the correct way to do it. I read quick and thought you were assigning an variable without spaces, which would have been a violation.

Reply:
I removed the coefficient label since it wasn't needed.  The reason it passed was that this is the PEP8 standard for keyword arguments (no space), even though it looks like an independent assignment line.